### ISSUE TITLE
incorrect type mismatch for  union typed array 

### ISSUE THREAD

**Issue by idoros** [facebook/flow#326]
```javascript
var numberAndStringArr:Array<number|string> = [1,2];
var stringArr:Array<string> = ['a','b'];

var result = numberAndStringArr.concat(stringArr);  // error
```

TypeScript handles this fine and assume output of `Array<number|string>`.
Is this the correct behavior?

**Comment by avikchaudhuri** [comment]
This is an issue with the way we wrote the signature of Array#concat, which currently demands the argument type of concat to be the same as that of the receiver. Fortunately, we introduced bounded polymorphism recently (http://flowtype.org/blog/2015/03/12/Bounded-Polymorphism.html) using which we can now revise the signature of Array#concat so that calling it on Array<T> with any Array<S> as argument works, as long as S is subtype of T. This should land in the next release.

**Comment by avikchaudhuri** [comment]
(Will close when next release lands.)

**Comment by jeffmo** [comment]
@idoros : Mind confirming this was fixed for you in 0.7.0?

**Comment by idoros** [comment]
fixed :) thanks
